main.go

```diff
@@ -40,16 +42,51 @@ func check() {
 	}
 }
 
-func perform() {
+func performStage1SelfUpdate(ctx context.Context) {
```

Comment:
So, if there is new version of pmm-update it updates it and fails with exit code 1? And then user have to click to update button again?

Reply:
1. `pmm-update -perform` exits with non-zero status.
2. supervisord notices that and restarts pmm-update: https://github.com/percona/pmm-server/blob/7d72185cb45905a6e63cea1b370223a2c6ce537a/supervisord.conf#L249-L250
3. `pmm-update -perform` tries to update itself again, notices that there is no update, and moves to the Ansible stage.

This and other ingenious bits will be in the Confluence next week.